pinot-segment-local/src/main/java/org/apache/pinot/segment/local/recordtransformer/DataTypeTransformer.java

```diff
@@ -128,6 +89,62 @@ public void transform(GenericRow record) {
     }
   }
 
+  /**
+   * Convert a single value to the target PinotDataType.
+   * This method can be reused by other components that need single value conversion.
+   *
+   * @param value The value to convert
+   * @param dest The target PinotDataType
+   * @param columnName The column name (for error reporting)
+   * @return The converted value
+   */
+  public static Object convertValue(@Nullable Object value, PinotDataType dest, String columnName) {
```

Comment:
(minor) `value` is never `null` (caller ensures that). Let's remove the `@Nullable` and null check. The return value could actually be `null`, so let's annotate the return value as `@Nullable`. Also suggest renaming `dest` to `targetType`

```suggestion
  @Nullable
  public static Object convertValue(Object value, PinotDataType targetType, String columnName) {
```